src/features/speedup/components/SpeedUpModal.tsx

```diff
@@ -0,0 +1,146 @@
+import { FEATURES } from '@/utils/chains'
+import { isWalletRejection } from '@/utils/wallets'
+import { type TransactionResponse } from 'ethers'
+
+type Props = {
+  open: boolean
+  handleClose: () => void
+  tx: TransactionResponse
+  txId: string
+  txHash: string
+  signerAddress: string | undefined
+  signerNonce: number | undefined | null
+}
+export const SpeedUpModal = ({ open, handleClose, tx, txId, txHash, signerAddress, signerNonce }: Props) => {
+  const [speedUpFee] = useGasPrice(true)
+  const [waitingForConfirmation, setWaitingForConfirmation] = useState(false)
+  const { gasLimit } = tx
+  const isEIP1559 = useHasFeature(FEATURES.EIP1559)
+
+  const wallet = useWallet()
+  const onboard = useOnboard()
+  const chainInfo = useCurrentChain()
+  const safeAddress = useSafeAddress()
+  const hasActions = signerAddress && signerNonce !== undefined && signerAddress === wallet?.address
+  const dispatch = useAppDispatch()
+
+  const isDisabled = waitingForConfirmation || !wallet || !speedUpFee || !onboard
+
+  const onSubmit = useCallback(async () => {
+    if (!wallet || !speedUpFee || !onboard || !chainInfo || !tx.to) {
+      return null
+    }
+
+    const txOptions = getTxOptions(
+      {
+        ...speedUpFee,
+        userNonce: signerNonce ?? undefined,
+        gasLimit: gasLimit,
+      },
+      chainInfo,
+    )
+
+    try {
+      setWaitingForConfirmation(true)
+      await dispatchTxSpeedUp(txOptions, txId, tx.to, tx.data, onboard, chainInfo?.chainId, safeAddress)
+
+      if (txHash) {
+        SimpleTxWatcher.getInstance().stopWatchingTxHash(txHash)
+      }
+
+      setWaitingForConfirmation(false)
+      handleClose()
+    } catch (e) {
+      const error = asError(e)
+      setWaitingForConfirmation(false)
+      if (!isWalletRejection(error)) {
+        dispatch(
+          showNotification({
+            message: 'Speed up failed',
+            variant: 'error',
+            detailedMessage: error.message,
+            groupKey: txHash,
+          }),
+        )
+      }
+    }
+  }, [
+    wallet,
+    speedUpFee,
+    onboard,
+    chainInfo,
+    signerNonce,
+    gasLimit,
+    safeAddress,
+    txHash,
+    handleClose,
+    dispatch,
+    tx,
+    txId,
+  ])
+
+  if (!hasActions) {
+    return null
+  }
+
+  return (
+    <ModalDialog open={open} onClose={handleClose} dialogTitle="Speed up transaction">
+      <DialogContent sx={{ p: '24px !important' }}>
+        <Box display="flex" justifyContent="center" alignItems="center" mb={2}>
+          <SvgIcon inheritViewBox component={RocketSpeedup} sx={{ width: 90, height: 90 }} />
+        </Box>
+        <Typography data-testid="speedup-summary">
+          This will speed up the pending transaction by{' '}
+          <Typography component={'span'} fontWeight={700}>
+            replacing
+          </Typography>{' '}
+          the original gas parameters with new ones.
+        </Typography>
+        <Box mt={2}>
```

Comment:
```suggestion
        </Box>
        
        <Typography data-testid="speedup-summary">
          This will speed up the pending transaction by{' '}
          <Typography component={'span'} fontWeight={700}>
            replacing
          </Typography>{' '}
          the original gas parameters with new ones.
        </Typography>
        
        <Box mt={2}>
```

A newline between blocks makes it easier to read.